### PR TITLE
gh-84443: SSLSocket.recv_into() shouldn't require __len__

### PR DESCRIPTION
**Pull request by ZackerySpytz**
https://bugs.python.org/issue40262

* Issue: gh-84443

### CODE REVIEW

**Review by tiran — APPROVED** [review]
LGTM

File: Lib/test/test_ssl.py [hunk 1]
```diff
@@ -3607,6 +3607,22 @@ def test_recv_zero(self):
...
+    def test_recv_into_buffer_protocol_len(self):
+        server = ThreadedEchoServer(CERTFILE)
+        server.__enter__()
+        self.addCleanup(server.__exit__, None, None)
+        s = socket.create_connection((HOST, server.port))
+        self.addCleanup(s.close)
+        s = test_wrap_socket(s, suppress_ragged_eofs=False)
+        self.addCleanup(s.close)
+
+        class B(bytearray):
```
Comment (serhiy-storchaka) [inline Lib/test/test_ssl.py:3534]: I think it would be better to test with more realistic example: `array('I', [0])`.

**Review by serhiy-storchaka — COMMENTED** [review]
What happen if keep `nbytes = None`?

**Comment by ambv** [comment]
This missed the boat for inclusion in Python 3.9 which accepts security fixes only as of today.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**